math/src/field/fields/montgomery_backed_prime_fields.rs

```diff
@@ -197,6 +199,32 @@ where
     }
 }
 
+impl<C, const NUM_LIMBS: usize>
+    Distribution<FieldElement<MontgomeryBackendPrimeField<C, NUM_LIMBS>>> for Standard
+where
+    C: IsMontgomeryConfiguration<NUM_LIMBS> + Clone + Debug,
+{
+    fn sample<R: Rng + ?Sized>(
+        &self,
+        rng: &mut R,
+    ) -> FieldElement<MontgomeryBackendPrimeField<C, NUM_LIMBS>> {
+        let mut rand_limbs: [u64; NUM_LIMBS] = [0; NUM_LIMBS];
+        let mut roll_to_max: bool = false;
+        let mod_min_one = C::MODULUS - UnsignedInteger::from_u64(1_u64);
+
+        for (i, rand_limb) in rand_limbs.iter_mut().enumerate().take(NUM_LIMBS) {
+            if !roll_to_max {
+                *rand_limb = rng.gen_range(0..=mod_min_one.limbs[i]);
+                roll_to_max = *rand_limb != mod_min_one.limbs[i];
+            } else {
+                *rand_limb = rng.gen_range(0..=u64::MAX);
+            }
+        }
```

Comment:
One last thing, this for would be easier if it's just a for i, and we can quickly check the endianess